### PR TITLE
Fix astar_avoid crashes

### LINKED ISSUE #4: Astar avoid bad_alloc error
<!--
FOR SUPPORT REQUESTS, please ask at ROS Answers: https://answers.ros.org/questions/ask/?tags=autoware, make sure to use the "autoware" tag.
For general discussion, please use the Autoware Discourse category: https://discourse.ros.org/c/autoware
Not sure if this is the right repository? Open an issue on https://gitlab.com/autowarefoundation/autoware.ai/autoware
For CONFIRMED bug reports, please fill out the information below.
IF YOUR BUG REPORT IS NOT CONFIRMED then you should consider posting to ROS Answers instead. You will get help faster.
Be as detailed as possible.
-->

### Required information:

- Operating system and version: 
  - Ubuntu 18.04
- Autoware installation type: 
  - Source (Script given by autonomouStuff)
- Autoware version or commit hash : 
  - `d392ba8d54fe4a22c89eb60c6f0884fa12348af4`
- ROS distribution and version: 
  - Melodic
- ROS installation type: 
  - (Script given by autonomouStuff)
- Package or library, if applicable: 
  - astar_avoid

### Description of the bug
<!-- Provide a clear and concise description of what the bug is. -->

Using LGSVL sim, astar avoid crashes in avoidance_mode when stopping.

### Steps to reproduce the bug

<!-- Detailed instructions on how to reliably reproduce the bug, e.g.
1. Launch the velodyne_driver using `ros2 launch velodyne_driver velodyne_node`
1. Launch rviz using `rviz`
1. Add the laser scan visualiser
1. ...
1. See the error displayed
Provide a short, self-contained, correct example: http://sscce.org/
If you include code, do not use a screenshot.
``` Provide code that can be copy-pasted ``` -->

 - Load map + get the car localized with ndt_matching
 - Load waypoints and put obstacles on the way
 - Detects objects with lidar_euclidean_cluster_detect + used them in costmap_generator_option to generate a costmap.
 - Start waypoint_loader, waypoint_replanner, waypoint_marker_publisher, lane_rule, lane_stop and lane_select.
 - Start astar_avoid in avoidance_mode, astar_navi, velocity_set, twist_filter and pure_pursuit.

Error message : 

```
[ INFO] [1582019551.725842839]: RELAYING -> STOPPING, Decelerate for stopping                                                                                                                                                                 

terminate called after throwing an instance of 'std::bad_alloc'                                                                                                                                                                               
  what():  std::bad_alloc                                                                                                                                                                                                                     

[astar_avoid-1] process has died [pid 7504, exit code -6, cmd /home/maxandre/autoware.ai/install/waypoint_planner/lib/waypoint_planner/astar_avoid costmap:=semantics/costmap_generator/occupancy_grid __name:=astar_avoid __log:=/home/maxand
re/.ros/log/e7f5412c-5230-11ea-9e82-9cb6d0c541d5/astar_avoid-1.log].                                                                                                                                                                          
log file: /home/maxandre/.ros/log/e7f5412c-5230-11ea-9e82-9cb6d0c541d5/astar_avoid-1*.log                                                                                                                                                     

```


### Expected behavior
<!-- A clear and detailed description of what you expected to happen. -->
The car should detect the obstacle, slow down, replan a new way around it, and follow it.


### Actual behavior
<!-- A clear and detailed description of what actually happens. -->
Astar_avoid crashes after an object is detected

You can find a rosbag of what is happening [here](https://drive.google.com/open?id=1LgLLyFEzz6LNjzARkBVbHDAvcqj4Hjib)

---

[Original bug report](https://gitlab.com/autowarefoundation/autoware.ai/core_planning/-/issues/18)

### PR DESCRIPTION
**Pull request by icolwell-as**
Must be merged after https://github.com/Autoware-AI/core_planning/pull/35

Resolves #4 I think (it did for our fork at least, might not be the same due to other changes made in upstream)

See original MR for more details: https://gitlab.com/astuff/autoware.ai/core_planning/-/merge_requests/76

### CODE REVIEW

File: waypoint_planner/src/astar_avoid/astar_avoid.cpp [hunk 1]
```diff
@@ -84,13 +76,37 @@ void AstarAvoid::currentVelocityCallback(const geometry_msgs::TwistStamped& msg)
 void AstarAvoid::baseWaypointsCallback(const autoware_msgs::Lane& msg)
 {
   base_waypoints_ = msg;
+
+  if (base_waypoints_initialized_)
+  {
+    // detect waypoint change by timestamp update
+    ros::Time t1 = prev_base_waypoints.header.stamp;
```
Comment (JWhitleyWork) [inline waypoint_planner/src/astar_avoid/astar_avoid.cpp:83]: `prev_base_waypoints` was not declared prior to this line.

**Comment by JWhitleyWork** [comment]
@icolwell-as Please correct the above issue so we can merge the dependent PRs.

**Comment by icolwell-as** [comment]
> @icolwell-as Please correct the above issue so we can merge the dependent PRs.

Need to merge #35 first, sorry for all these chained PRs! :smile:

**Comment by JWhitleyWork** [comment]
Kicking CI now that #35 is merged.

**Comment by JWhitleyWork** [comment]
@icolwell-as Conflicts. :)

**Pull request merged**